commcare_connect/opportunity/helpers.py

```diff
@@ -1,42 +1,58 @@
 from django.db.models import Case, Count, F, Max, Min, Q, Sum, When
 
-from commcare_connect.opportunity.models import Opportunity, OpportunityAccess, VisitValidationStatus
+from commcare_connect.opportunity.models import Opportunity, OpportunityAccess, UserInvite, VisitValidationStatus
 
 
 def get_annotated_opportunity_access(opportunity: Opportunity):
     learn_modules_count = opportunity.learn_app.learn_modules.count()
     access_objects = (
-        OpportunityAccess.objects.filter(opportunity=opportunity)
-        .select_related("user", "opportunityclaim")
+        UserInvite.objects.filter(opportunity=opportunity)
+        .select_related("opportunity_access", "opportunity_access__opportunityclaim", "opportunity_access__user")
         .annotate(
-            last_visit_date_d=Max("user__uservisit__visit_date", filter=Q(user__uservisit__opportunity=opportunity)),
+            last_visit_date_d=Max(
+                "opportunity_access__user__uservisit__visit_date",
+                filter=Q(opportunity_access__user__uservisit__opportunity=opportunity),
```

Comment:
Should this exclude trial visits?

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/305/commits/501472374aac7b65bf31b50b88289bb05a519d8a